cms/djangoapps/contentstore/views/tabs.py

```diff
@@ -110,54 +134,52 @@ def reorder_tabs_handler(course_item, request):
     try:
         CourseTabList.validate_tabs(new_tab_list)
     except InvalidTabsException as exception:
-        return JsonResponse(
-            {"error": f"New list of tabs is not valid: {str(exception)}."}, status=400
-        )
+        raise ValidationError(
+            {"error": f"New list of tabs is not valid: {str(exception)}."}
+        ) from exception
 
     # persist the new order of the tabs
     course_item.tabs = new_tab_list
-    modulestore().update_item(course_item, request.user.id)
-
-    return JsonResponse()
+    modulestore().update_item(course_item, user.id)
 
 
-def edit_tab_handler(course_item, request):
+def edit_tab_handler(course_item: CourseBlock, tabs_data: Dict, user: User):
     """
     Helper function for handling requests to edit settings of a single tab
     """
 
     # Tabs are identified by tab_id or locator
-    tab_id_locator = request.json['tab_id_locator']
+    tab_id_locator = tabs_data['tab_id_locator']
 
     # Find the given tab in the course
     tab = get_tab_by_tab_id_locator(course_item.tabs, tab_id_locator)
     if tab is None:
-        return JsonResponse(
-            {"error": f"Tab with id_locator '{tab_id_locator}' does not exist."}, status=400
-        )
-
-    if 'is_hidden' in request.json:
-        # set the is_hidden attribute on the requested tab
-        tab.is_hidden = request.json['is_hidden']
-        modulestore().update_item(course_item, request.user.id)
-    else:
-        raise NotImplementedError(f'Unsupported request to edit tab: {request.json}')
+        raise ValidationError({"error": f"Tab with id_locator '{tab_id_locator}' does not exist."})
 
-    return JsonResponse()
+    if 'is_hidden' in tabs_data:
+        if tab.is_hideable:
+            # set the is_hidden attribute on the requested tab
+            tab.is_hidden = tabs_data['is_hidden']
+            modulestore().update_item(course_item, user.id)
+        else:
+            raise ValidationError({"error": f"Tab of type {tab.type} can't be hidden"})
```

Comment:
`can't` => `can not`

Reply:
you are mixing double quotes and singles quotes here and there, it should be consistent. pep8 states to pick a rule and stick with it. I will get more information about it what is edx expectation of using single vs double-quotes. I suspect it's single quotes.

Reply:
Note that I haven't written most of the code here. I am reusing legacy code, so I have inherited most of the existing format issues as you can see here: https://github.com/edx/edx-platform/blob/0021be0a71fe2469042259ff522aff4e17734184/cms/djangoapps/contentstore/views/tabs.py#L124-L146

I generally avoid making too many formatting changes to existing code so as to avoid unnecessary diffs. 

That said, for all new code I try to format it consistently. I don't think there is currently a standardised style for quotes for edx-platform currently that I am aware off. 

There was an effort long ago to standardise on using `black` as the autoformatter: 
- https://discuss.openedx.org/t/transitioning-towards-autoformatting/3807
- https://github.com/edx/open-edx-proposals/pull/107

Black uses double-quotes, so I use that for all new code to avoid future unnecessary reformatting.

Reply:
Makes sense let's use double-quotes 👍 .